app/services/db/sp_cost/add_sp_cost.rb

```diff
@@ -21,14 +21,14 @@ class SpCostTypeError < StandardError; end
       ].freeze
 
       def self.call(issuer, ial, token)
-        return if issuer.nil? || token.blank?
+        return if token.blank?
         unless TOKEN_WHITELIST.include?(token.to_sym)
           NewRelic::Agent.notice_error(SpCostTypeError.new(token.to_s))
           return
         end
-        sp = ServiceProvider.find_by(issuer: issuer)
+        sp = issuer.blank? ? nil : ServiceProvider.find_by(issuer: issuer)
         agency_id = sp ? sp.agency_id : 0
```

Comment:
I think we could remove this intermediate assignment to `sp` and all the extra ternaries and just get `agency_id` right?

```suggestion
        agency_id = (issuer.present? && ServiceProvider.find_by(issuer: issuer)&.agency_id) || 0
```

Reply:
I like!